### DISCUSSION TITLE
ErrorBoundary when I use field array with watch

### DISCUSSION THREAD

**Discussion by sabrynnall** [react-hook-form/react-hook-form#9418] [Q&A]
Hi, guys! Every time I use watch to track my field array, I get the following error:

> React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.

This happens in the following snippet:

```
const { append, remove, fields, insert } = useFieldArray({
    control,
    name: 'requirements',
  });

  const watchFieldArray = watch('requirements');

  const controlledFields = fields.map((field, index) => {
    return {
      ...field,
      ...watchFieldArray[index]
    };
  });
```

Can someone help me?

**Comment by bluebill1049** [comment]
can you share an actual usage in a codesandbox?
